crates/cairo-coverage/src/input/filter.rs

```diff
@@ -76,15 +86,23 @@ impl Filter {
             labels.insert(ItemLabel::Macro);
         }
 
-        // TODO(#55)
-        // TODO: We should probably filter by path to user project not by path to cache
-        // TODO: Can get this from source_sierra_path in call trace
-        if file_location.contains("com.swmansion.scarb") || file_location.contains(".cache/scarb") {
-            labels.insert(ItemLabel::NonUserFunction);
-        } else {
+        if file_location.contains(&self.user_project_path) {
             labels.insert(ItemLabel::UserFunction);
+        } else {
+            labels.insert(ItemLabel::NonUserFunction);
         }
 
         labels
     }
 }
+
+fn find_user_project_path(source_sierra_path: &str) -> Result<String> {
```

Comment:
This function seems to be trying to find a project root, right? why can't we just take this info from scarb metadata? (Sorry if it's a dumb question, maybe I'm missing some crucial info about coverage)

Reply:
Technically `cairo-coverage` command can be run from any location, so attempting to find metadata for cwd may fail. We would need to know the path to the manifest file.

Reply:
Ok, this seems like a valid reason, thanks. 

Just to make sure I understand things correctly - So we require users to supply us with call_trace_path, that holds the info about sierra path right? So technically, we can never be wrong when we try to determine the path itself. By the name of "SNFORGE_SIERRA_DIR" I assume this (and the layout underneath) is something we do have a strict control over, so this ideally should never change out of the blue. 

If that's true, I guess it's fine to have this not be dependent on scarb. I do, however, think we should think about some kind of test for this function, so we immediately know if something changes in snforge regarding the path.

Reply:
> Just to make sure I understand things correctly - So we require users to supply us with call_trace_path, that holds the info about sierra path right?

Correct

> So technically, we can never be wrong when we try to determine the path itself. By the name of "SNFORGE_SIERRA_DIR" I assume this (and the layout underneath) is something we do have a strict control over, so this ideally should never change out of the blue

This it true for now given that `snforge` is the only tool that can produce trace files. We need to pay attention to this when integrating with `cairo-test` / `cairo-run` to follow the same layout.

Reply:
1. This is tested in every test implicitly. As we generate traces with `snforge` in tests, if the layout changes, all tests will fail because they will include `std` code and the output won't match.

2. This is not an ideal solution, but relying on `scarb` metadata isn't either, as you need to directly depend on `scarb`. The best way to handle this, in my opinion, is to add a boolean flag to mappings. We can do this because the compiler knows the list of crate IDs of the currently compiled project. There is a planned process to design more ergonomic and overall better mappings (https://github.com/software-mansion/cairo-coverage/issues/44).

Reply:
ad 2. I think this project already depends on scarb, but I get the gist. Indeed it seems like the proposed task solves the issue better. Thanks for giving me more context on this, you can resolve the conversation :+1: